org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/translator/A4Solution.java

```diff
@@ -313,24 +323,31 @@
         TupleSet stringBounds = factory.noneOf(1);
         final TupleSet next = factory.noneOf(2);
         int min = min(), max = max();
-        if (max >= min)
-            for (int i = min; i <= max; i++) { // Safe since we know 1 <=
-                                              // bitwidth <= 30
-                Tuple ii = factory.tuple("" + i);
+        if (max >= min){
+            ConcurrentSkipListSet<Integer> ints = new ConcurrentSkipListSet<Integer>();
```

Comment:
Why `ConcurrentSkipListSet`?  It looks there are no concurrent accesses to `ints`

Reply:
good point! Changed to TreeSet.